starky/src/fibonacci_stark.rs

```diff
@@ -0,0 +1,98 @@
+use std::marker::PhantomData;
+
+use plonky2::field::extension_field::{Extendable, FieldExtension};
+use plonky2::field::packed_field::PackedField;
+use plonky2::hash::hash_types::RichField;
+use plonky2::plonk::circuit_builder::CircuitBuilder;
+
+use crate::constraint_consumer::{ConstraintConsumer, RecursiveConstraintConsumer};
+use crate::stark::Stark;
+use crate::vars::{StarkEvaluationTargets, StarkEvaluationVars};
+
+/// Toy STARK system used for testing.
+/// Computes a Fibonacci sequence with inital values `x0, x1` using the transition
+/// `x0 <- x1, x1 <- x0 + x1`.
+pub struct FibonacciStark<F: RichField + Extendable<D>, const D: usize> {
+    x0: F,
+    x1: F,
+    _phantom: PhantomData<F>,
+}
+
+impl<F: RichField + Extendable<D>, const D: usize> FibonacciStark<F, D> {
+    const NUM_COLUMNS: usize = 2;
+    const NUM_ROWS: usize = 1 << 5;
+
+    fn new(x0: F, x1: F) -> Self {
+        Self {
+            x0,
+            x1,
+            _phantom: PhantomData,
+        }
+    }
+
+    fn generate_trace(&self) -> Vec<[F; Self::NUM_COLUMNS]> {
+        (0..Self::NUM_ROWS)
+            .scan([self.x0, self.x1], |acc, _| {
+                let tmp = *acc;
+                acc[0] = tmp[1];
+                acc[1] = tmp[0] + tmp[1];
+                Some(tmp)
+            })
+            .collect()
+    }
+}
+
+impl<F: RichField + Extendable<D>, const D: usize> Stark<F, D> for FibonacciStark<F, D> {
+    const COLUMNS: usize = Self::NUM_COLUMNS;
+    const PUBLIC_INPUTS: usize = 0;
```

Comment:
It might be good to exercise public input logic too - we could set this to 1 and add

```rust
yield_constr.one_last_row(vars.local_values[1] - vars.public_inputs[0]);
```

Reply:
I'll add 3 public inputs, 2 for the starting state, and one for the second element of the last state.
The prover will have to take a new parameter `pis: [F; S::PUBLIC_INPUTS]` though, right?